server/trucks.js

```diff
@@ -253,4 +253,15 @@ var foodTrucks = [
  * that method should take a single parameter that represents the day to filter on
  * use underscore's filter() method to return all trucks that have the day in their
  * schedule
- * /
\ No newline at end of file
+ */
+var undsc = require('../node_modules/underscore/underscore');
+
+var filterByDay = function (day) {
+    var trucks = undsc.filter(foodTrucks, function (day) {
+        foodTrucks.schedule == day;
```

Comment:
First thing: don't use ==. There's never a reason to use == instead of ===. Second, you're close here - filter takes a function as it's second argument that should return true or false. If it returns true, the item is returned by filter, if it returns false, it's "filtered out". In this case, what you're trying to determine is if "day" is in the truck's schedule array. So the code is going to look something more like this:

`var trucks = undsc.filter(foodTrucks, function (truck) {
    return undsc.contains(truck.schedule, day);
});
return trucks;`

The function being passed as the second argument to filter has a parameter called 'truck', not 'day', because that function is getting passed a truck object as filter loops through the foodTrucks array. Inside filter, another underscore function, contains, is being called. That function takes an array and the thing to look for in the array. If the array contains the thing, it returns true, otherwise it returns false.